pysal/weights/Distance.py

```diff
@@ -489,27 +489,27 @@ def _distance_to_W(self, ids=None):
         neighbors = dict([(i,[]) for i in ids])
         weights = dict([(i,[]) for i in ids])
         if self.binary:
-            for key,weight in self.dmat.items():
-                i,j = key
+            for idxs,weight in self.dmat.items():
+                idx,jdx = idxs
+                i, j = ids[idx], ids[jdx]
```

Comment:
This is the crux. Normally, a sparse matrix's `.items()` method returns a list of tuples with indexes & a value at the index. 

To correctly stuff this into a named/id'd weights & neighbors dictionaries, we need to index into the collection of names/ids.